controllers/userStatus.js

```diff
@@ -103,7 +103,8 @@ const updateUserStatus = async (req, res) => {
   try {
     const userId = getUserIdBasedOnRoute(req);
     if (userId) {
-      const dataToUpdate = req.body;
+      const joinData = await getJoinData(userId);
+      const dataToUpdate = { ...req.body, monthlyHours: { committed: 4 * joinData[0].intro.numberOfHours } };
```

Comment:
this controller will update the no of hours every time from the join data. if I am not wrong this is suppose to happen only for the 1st time when they join. Right ?